src/store/entityDetailReducer.js

```diff
@@ -1,64 +1,35 @@
 import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
-import { Link } from 'react-router-dom';
-import React from 'react';
-import { ExpandedRow } from '../Components/RosTable/ExpandedRow';
-import { ProgressScoreBar } from '../Components/RosTable/ProgressScoreBar';
-
-export const systemName = (displayName, id, { inventory_id: inventoryId }) => {
-    return (
-        <Link to={{ pathname: `/${inventoryId}` }} className={ `pf-link system-link link-${inventoryId}` }>
-            { displayName }
-        </Link>
-    );
-};
-
-export const scoreProgress = (data) => {
-    return (
-        <ProgressScoreBar measureLocation='outside' valueScore={data} />
-    );
-};
-
-export const recommendations = (data, id, { inventory_id: inventoryId }) => {
-    return (
-        <Link to={{ pathname: `/${inventoryId}` }}
-            className={ `pf-link recommendations ${data === 0 ? 'green-400' : ''} link-${inventoryId}` }>
-            { data }
-        </Link>
-    );
-};
-
-const addExpandedView = (rowData) => {
-    const {
-        id, cloud_provider: cloudProvider, instance_type: instanceType,
-        idling_time: idlingTime, io_wait: ioWait
-    } = rowData;
-    return (<ExpandedRow { ...{ id, cloudProvider, instanceType, idlingTime, ioWait } } />);
-};
-
-function modifyInventory(columns, state) {
+import Recommendations from '../Components/Recommendations';
 
+const defaultState = { loaded: false };
+function entityLoaded(rosDetails, state) {
     return {
         ...state,
-        columns,
-        rows: state.rows.map((row) => ({
-            ...row,
-            children: addExpandedView(row)
-        })),
-        loaded: true
+        loaded: true,
+        activeApps: [
+            {
+                title: 'Recommendations',
+                name: 'ros_recommendations',
+                component: Recommendations
```

Comment:
@karelhala,
For component `Recommendations`, may I pass rosDetails as props? or I should make a request inside component itself? Any recommended way while adding App components?

Reply:
I think the safest way would be to refetch the data. Actually it might be fruitfull to create new reducer which will consume these data and you can re-use that reducer in `Recommendations` component and in `Inventory` header.

Reply:
+1, added separate reducer to fetch ROS system information.